happi/audit.py

```diff
@@ -59,6 +61,55 @@ def check_name_match_id(result: SearchResult) -> None:
     assert id_field == name_field, f'id: {id_field} != name: {name_field}'
 
 
+def check_wait_connection(result: SearchResult) -> None:
+    """
+    Check if all the signals on the device can connect to their PV's
+
+    This can take a long time, consider filtering your query before running
+    """
+    dev = result.get()
+
+    try:
+        dev.wait_for_connection(timeout=5)
```

Comment:
`wait_for_connection` is an ophyd-(presumably only v1)-ism. `happi audit` could potentially be used by others that don't use happi for ophyd devices. 

Should we have a `hasattr` check for `wait_for_connection`, or is there a better way to deselect this check for those users?

Reply:
Good point.  That was my reasoning for leaving out my lightpath audit function earlier, but I failed to remember that here.  I do think the `wait_for_connection` audit will be important for us to have deployed so I do want to include it.

My initial impression is that checking the database for ophyd-compatible devices would be a time-consuming process, since that's only known after instantiation.  I also wouldn't want to do something like abort the audit process if the method doesn't exist, since happi could host a mixture of device types. 

I think it's reasonable to simply let this check fail if the method doesn't exist (though it does that already, just not gracefully)  [d6074a4](https://github.com/pcdshub/happi/pull/295/commits/d6074a43559a53c048af99cb25e82c0f644a94b3)